bin/ch-dir2squash

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
+set -e
+
+libexec="$(cd "$(dirname "$0")" && pwd)"
+. "${libexec}/base.sh"
+
+# shellcheck disable=SC2034
+usage=$(cat <<EOF
+Pack a filesystem directory into a squashfs.
+
+Usage:
+
+  $ $(basename "$0") INDIR OUTDIR [ARGS ...]
+EOF
+)
+
+sentinel=WEIRD_AL_YANKOVIC
+
+parse_basic_args "$@"
+
+if [ $# -lt 2 ]; then
+    usage
+fi
+
+indir=$1
+outdir=$2
+
+shift 2
+
+# Check input directory exists
+if [ ! -e "$indir" ]; then
+    echo "can't squash: ${indir} does not exist" 1>&2
+    exit 1
+fi
+# Check input directory is a directory
+if [ ! -d "$indir" ]; then
+    echo "can't squash: ${indir} is not a directory" 1>&2
+    exit 1
+fi
+# Check input directory is a charliecloud image
+if [ ! -e "${indir}/${sentinel}" ]; then
+    echo "${indir} is NOT a Charliecloud iamge" 1>&2
+    exit 1
+fi
+
+# Check OUTDIR exists and is a directory
+if [ ! -e "$outdir" ]; then
+    echo "can't squash: ${outdir} does not exist" 1>&2
+    exit 1
+fi
+if [ ! -d "$outdir" ]; then
+    echo "can't squash: ${outdir} is not a directory" 1>&2
```

Comment:
Remove the brackets from variables that are not concatenated together or difficult to distinguish from the rest of the string. `${outdir}` should be `$outdir`.

Reply:
I was under the impression if the variable is involved in string substitution or concatenation it should be in brackets. Can you link me where I can find the rule to use on that? I'll have to go back and make it consistent in the other files.

Reply:
We generally use brackets for concatenation, which most commonly occurs in paths, e.g., `"${foo}/${bar}/fizz/buzz"`. If this same string was separated by spaces it would be:  
`"$foo $bar fizz buzz"`

Reply:
Here is another example:

```
docker_ok () {
    docker_tag_p "$1"
    docker_tag_p "${1}:latest"
    docker_tag_p "${1}:$(ch-run --version |& tr '~+' '--')"
}
```

Reply:
I tried to follow how it was done in the other helper scripts. 

in `ch-tar2dir` we have


`# Complain clearly if destination not valid (issue #340).`
`if [ ! -e "${2}" ]; then`
`    echo "can't unpack: ${2} does not exist" 1>&2`
 `   exit 1`
`fi`
`if [ ! -d "${2}" ]; then`
`    echo "can't unpack: ${2} is not a directory" 1>&2`
`    exit 1`
`fi`